test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Models/PackageItemListViewModelTests.cs

```diff
@@ -37,5 +58,401 @@ public void LocalSources_PackageReader_NotNull()
+            using (var testDir = TestDirectory.Create())
+            {
+                var imagePath = Path.Combine(testDir, "image.png");
+                CreateNoisePngImage(path: imagePath);
+
+                var packageItemListViewModel = new PackageItemListViewModel()
+                {
+                    IconUrl = new Uri(imagePath, UriKind.Absolute)
+                };
+
+                // Act
+                BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
+
+                // Assert
+                VerifyImageResult(result);
+                Assert.NotSame(DefaultPackageIcon, result);
+            }
+        }
+
+        [Theory]
+        [InlineData(@"/")]
+        [InlineData(@"\")]
+        public void IconUrl_EmbeddedIcon_RelativeParentPath_ReturnsDefault(string separator)
+        {
+            using (var testDir = TestDirectory.Create())
+            {
+                // Create decoy nuget package
+                var zipPath = Path.Combine(testDir.Path, "file.nupkg");
+                CreateDummyPackage(zipPath);
+
+                // prepare test
+                UriBuilder builder = new UriBuilder(new Uri(zipPath, UriKind.Absolute))
+                {
+                    Fragment = $@"..{separator}icon.png"
+                };
+
+                var packageItemListViewModel = new PackageItemListViewModel()
+                {
+                    IconUrl = builder.Uri,
+                    PackageReader = new Func<PackageReaderBase>(() => new PackageArchiveReader(zipPath))
+                };
+
+                // Act
+                BitmapSource result = GetFinalIconBitmap(packageItemListViewModel).Result;
+
+                // Assert
+                VerifyImageResult(result);
+                Assert.Same(DefaultPackageIcon, result);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(EmbeddedTestData))]
+        public void IsEmbeddedIconUri_Tests(Uri testUri, bool expectedResult)
+        {
+            var result = PackageItemListViewModel.IsEmbeddedIconUri(testUri);
+            Assert.Equal(expectedResult, result);
+        }
+
+        public static IEnumerable<object[]> EmbeddedTestData()
+        {
+            Uri baseUri = new Uri(@"C:\path\to\package");
+            UriBuilder builder1 = new UriBuilder(baseUri)
+            {
+                Fragment = "    " // UriBuilder trims the string
+            };
+            UriBuilder builder2 = new UriBuilder(baseUri)
+            {
+                Fragment = "icon.png"
+            };
+            UriBuilder builder3 = new UriBuilder(baseUri)
+            {
+                Fragment = @"..\icon.png"
+            };
+            UriBuilder builder4 = new UriBuilder(baseUri)
+            {
+                Fragment = string.Empty // implies that there's a Tag, but no value
+            };
+            UriBuilder builder5 = new UriBuilder(baseUri)
+            {
+                Query = "aParam"
+            };
```

Comment:
Use `var` since the right-hand sides are `new` expressions.

Reply:
yep fixed in next iteration ... all the uribuilders.